subsys: controller: bluetooth add check for NullPointer

Majistor opened the pull request:
Added a check for null pointer in ull_sync_iso.c file to avoid potential crashes.

Fixes: #81997

cvinayak reviewed (changes requested):
Update commit title to “Bluetooth: Controller: Fix missing null pointer check”

https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines

Also, refer to the suggested change as here: https://github.com/zephyrproject-rtos/zephyr/issues/81997#issuecomment-3317374424

non-compliant inline comments and added code is not needed.

refer to, for review rework here 
https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow

cvinayak commented:
There are two commits, squash them and force push…

ljd42 commented on subsys/bluetooth/controller/ll_sw/ull_sync_iso.c:
```diff
@@ -584,6 +584,8 @@ void ull_sync_iso_setup(struct ll_sync_iso_set *sync_iso,
 			      lll->window_size_event_us;
 	/* Skip to first selected BIS subevent */
 	stream = ull_sync_iso_stream_get(lll->stream_handle[0]);
+LL_ASSERT(stream);
```
please make sure that the `LL_ASSERT` is properly indented at the same level as the line before/after your change, see below.

While that formatting doesn't modify the end result (the resulting binary will be the same), it allows us humans to quickly grasp which code belong to which if/else branches.

Currently, your change looks like this:
```c
        /* Skip to first selected BIS subevent */
        stream = ull_sync_iso_stream_get(lll->stream_handle[0]);
LL_ASSERT(stream);
  
        if (lll->bis_spacing >= (lll->sub_interval * lll->nse)) {
                sync_iso_offset_us += (stream->bis_index - 1U) *
                                      lll->sub_interval *
                                      ((lll->irc * lll->bn) + lll->ptc);
        } else ... 
```
This should look like this:
```c
        /* Skip to first selected BIS subevent */
        stream = ull_sync_iso_stream_get(lll->stream_handle[0]);
        LL_ASSERT(stream);
  
        if (lll->bis_spacing >= (lll->sub_interval * lll->nse)) {
                sync_iso_offset_us += (stream->bis_index - 1U) *
                                      lll->sub_interval *
                                      ((lll->irc * lll->bn) + lll->ptc);
        } else ... 
```

Majistor replied:
I did the same earlier, but my commit was failing on the compliance test.

Majistor replied:
The latest commit seems to be perfectly indented on my editor, but it's seems to be left shifted in the commit and then also failing the compliance test

ljd42 reviewed (changes requested):
Please properly indent your change, see my comment.

Also,  squash all your commits into a single commit using `git rebase`. Let me  know if you need help with this step.

Majistor committed:
Bluetooth: Controller: Fix missing null pointer check

Added a check for null pointer in ull_sync_iso.c file
to avoid potential crashes.

Fixes: #81997

Signed-off-by: Ashirwad Paswan <ashi06712@gmail.com>

ljd42 reviewed (approved):
LGTM.  Thanks for your contribution!
